domtemplate.php

```diff
@@ -123,12 +101,35 @@ abstract class DOMTemplateNode {
 	// html_entity_decode : convert HTML entities back to UTF-8
 	//--------------------------------------------------------------------------
 	public function html_entity_decode ($html) {
+		// a table of HTML entites to reverse:
+		// '&', '<', '>' are removed so we don’t turn user text into working HTML!
+		//
+		// TODO: moving DOMTemplate to a namespace will allow us to use
+		// 		 a private variable for the namespace, rather than per-instance
+		//		 as generating this table is slow
+		static $htmlentities = null;
+
+		// construct the HTML entities table:
+		// this function was added in PHP 5.3.4, but the extended HTML5
+		// entities list was not added until later (don't know which version)
+		//
+		// the full list of HTML5 entities, as per the spec, is listed here:
+		// <https://html.spec.whatwg.org/multipage/named-characters.html>
+		//
+		if (is_null($htmlentities)) {
+			$htmlentities = array_flip (array_diff (
+				get_html_translation_table(
+					HTML_ENTITIES, ENT_NOQUOTES | ENT_HTML401, 'UTF-8'
+				),
+				["&amp;", "&lt;", "&gt;"]
+			));
+		}

```

Comment:
Now the list generation would run maximum twice, once for each extending class first invocation. Making this function a standalone one (as you note in a comment above it) would solve it.

Reply:
The next version of DOMTemplate will be switching to using PHP namespaces, so a namespace-private constant / function can be used, instead of stuffing it into a class property, as was the tradition for PHP 5.2+ code.

Reply:
That would be a better way to handle it surely. I didn’t want to make a big change, just wanted to make it work again.